### ISSUE TITLE
Some Confusion About Algorithmic Predictions

### ISSUE THREAD

**Issue by CHENHUI-X** [dmlc/xgboost#9211]
Hello developer, I may not know much about the GBDT algorithm, please forgive me for asking such a stupid question. What I want to ask is:

- We know that the GBDT or XGB algorithm has a learning rate parameter during training, so that the final output $f_{t}(x) = f_{t-1}(x) + lr * h_{t}(x)$, Then what I want to ask is, when predicting, for the sample x used, his prediction result is still $f(x) = h_{0}(x) + lr * h_{1}(x) + lr * h_{2}(x) + ... + lr * h_{t}(x)$ is the final output?

- From a logical point of view, the above seems to be true, but I saved the trained model, checked its results, and then manually calculated a sample x, and found that the final result seemed to simply put this sample x in each tree The leaf node values ​​are added up. (The code is at the bottom)

This confuses me. Did I not catch up somewhere, or when you saved the model, you have automatically scaled the leaf nodes so that the final output can be simply summed directly.
```python

  import xgboost as xgb
  from sklearn.datasets import make_classification
  from sklearn.model_selection import train_test_split
  # # Load data : binary class
  X , y = make_classification(n_samples = 100 , n_features = 6 , n_classes = 2 , n_clusters_per_class = 1 ,
                              random_state = 42)
  X_train , X_test , y_train , y_test = train_test_split(X , y , test_size = 0.3 , random_state = 42)
  
  data = X_train
  label = y_train
  dtrain = xgb.DMatrix(data , label = label)
  data = X_test
  label = y_test
  dtest = xgb.DMatrix(data , label = label)
  
  param = { 'max_depth' : 2
      , 'eta' : 0.1
      , 'objective' : 'binary:logistic'
      , 'eval_metric' : 'auc' }
  
  evallist = [(dtrain , 'train') , (dtest , 'eval')]
  num_round = 2
  bst = xgb.train(param , dtrain , num_round , evallist , evals = dtest , early_stopping_rounds = 10)
  
  y_pred_pro = bst.predict(xgb.DMatrix(X_test[:2,:] )) #, iteration_range = (0 , bst.best_iteration + 1)
  print(X_test[:2,:])
  print(y_pred_pro)
  modelLoc = 'xgb_model_file'
  bst.dump_model(open(modelLoc ,'w'))
```

- Here is the trees 
```
booster[0]:
0:[f1<-0.269732147] yes=1,no=2,missing=1
	1:[f0<0.0448073633] yes=3,no=4,missing=3
		3:leaf=-0
		4:leaf=-0.180487797
	2:leaf=0.175757572
booster[1]:
0:[f1<-0.269732147] yes=1,no=2,missing=1
	1:[f0<0.0448073633] yes=3,no=4,missing=3
		3:leaf=-0
		4:leaf=-0.165453374
	2:leaf=0.161441877

```

- For samples [0.20613852, 0.75783672, -0.89841467, 0.67963113, 0.49191917, 0.42265275], the official predict function result is 0.5835101, which seems to be the sum of the leaf nodes: 0.17575 7572, 0.161441877, and then the result after the sigmiod function.

- However, in the [GBDT algorithm implemented by sklearn](https://github.com/scikit-learn/scikit-learn/blob/364c77e047ca08a95862becf40a04fe9d4cd2c98/sklearn/ensemble/_gradient_boosting.pyx#LL89C27-L89C33)  , when predicting, the final result still seems to be the result of each class tree multiplied by the learning rate, and then summed to get the final result.

**Comment by trivialfis** [comment]
> or when you saved the model, you have automatically scaled the leaf nodes so that the final output can be simply summed directly.

Yup. The leaf is scaled by learning rate.
